bazel/test.bzl

```diff
@@ -434,46 +427,43 @@ def redpanda_cc_bench(
         data = data,
     )
 
+    # below here targets use a wrapper which takes the binary as the first
+    # argument and which should be included in the runfiles data
     args = ["$(rootpath :{})".format(binary_name)] + args
+    data = data + [":" + binary_name]
 
     env = env | {
         "MB_EXEC_IN_SHM": "1" if exec_in_shm else "0",
         "MB_REDIRECT_STDERR_DEFAULT": "1" if redirect_stderr else "0",
     }
 
+    wrapper_src = ["//tools:bench_wrapper"]
+
+    binary_args = []
+    if runs != None:
+        binary_args.append("--runs={}".format(runs))
+    if duration != None:
+        binary_args.append("--duration={}".format(duration))
+
     # to run a benchmark in the right way, we need to wrap it in bench-wrapper.sh,
     # which can cd to the right location and make other adjustments
     native.sh_binary(
         name = name,
-        srcs = ["//tools:bench_wrapper"],
+        srcs = wrapper_src,
         args = args,
-        data = data + [
-            ":" + binary_name,
-        ],
+        data = data,
         env = env,
         testonly = True,
     )
 
-    # we write a wrapper to test the benchmark, which tries to
-    # run it as quickly as possible in order to smoke test it
-    write_file(
-        name = name + "_test_script",
-        out = name + "_test_wrapper.sh",
-        content = [
-            "#!/bin/bash",
-            "exec $@ --iterations=1 --runs=1 --duration=0 --no-stdout --overprovisioned",
-        ],
-    )
     test_data, test_env = _test_options()
     native.sh_test(
         name = name + "_test",
         timeout = timeout,
         tags = resource_tags + tags,
-        srcs = [name + "_test_script"],
+        srcs = wrapper_src,
         env = env | test_env,
-        args = args,
-        data = [
-            ":" + binary_name,
-        ] + data + test_data,
+        args = args + ["--iterations=1 --runs=1 --duration=0 --no-stdout --overprovisioned"],
```

Comment:
I think you want these as separate strings.

Reply:
(more than noise, doing it gives the misleading impression that each arg will be passed as a whole arg, even if it has spaces, which is not the case)

Reply:
yeah I was under the impression they would not be `shlex.split` but yeah fine as is.